File: sololearn/tasks/binary_search.js
function binarySearch(arr, target) {
  let l = 0, r = arr.length - 1;
  while (l <= r) {
    const m = Math.floor((l + r) / 2);
    if (arr[m] === target) return m;
    arr[m] < target ? l = m + 1 : r = m - 1;
  }
  return -1;
}

File: sololearn/lua_snippets/binary_search.lua
-- Task: Implement binary search on sorted array
-- Tests algorithmic thinking and edge cases

function binarySearch(arr, target)
  local left, right = 1, #arr

  while left <= right do
    local mid = math.floor((left + right) / 2)
    if arr[mid] == target then
      return mid
    elseif arr[mid] < target then
      left = mid + 1
    else
      right = mid - 1
    end
  end

  return -1
end
